app/components/SignupForm.tsx

```diff
@@ -1,35 +1,179 @@
 "use client";
-import React from "react";
 
-const SignupForm: React.FC = () => {
-  const handleSubmit = async (e: React.FormEvent) => {
+import React, { useState } from "react";
+
+const SignupForm = () => {
+  const [form, setForm] = useState({
+    id: "",
+    fullname: "",
+    email: "",
+    password: "",
+    confirm_password: "",
+  });
+
+  const [idError, setIdError] = useState("");
+  const [fullnameError, setFullnameError] = useState("");
+  const [emailError, setEmailError] = useState("");
+  const [passwordError, setPasswordError] = useState("");
+  const [confirmPasswordError, setConfirmPasswordError] = useState("");
+
+  const handleChange = (e) => {
```

Comment:
```suggestion
  const handleChange = (e: : React.ChangeEvent<HTMLInputElement>) => {
```
type이 빠져있습니다. 암묵적인 any 타입 사용을 지양해주세요!